### PR TITLE
Fix for scale above precision 38

### PR DESCRIPTION
**Pull request by mithilagarud-dm**
Fix for scale above precision 38

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java [hunk 1]
```diff
@@ -2452,11 +2454,17 @@ public static String getDataTypeBasedOnPrecision(int precision, int scale)  {
     }
   }
 
-  private static int adjustPrecisionIfNeeded(String dataType, int precision) {
-    if (dataType.equals("BIGNUMERIC") && precision > 38) {
-      return 38;
+  private static int[] adjustPrecisionAndScaleIfNeeded(String dataType, int precision, int scale) {
+    int[] precisionScale = {precision, scale};
+    if (dataType.equals("BIGNUMERIC")) {
+      if (precision > 38) {
```
Comment (sauajmera) [inline core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java:2460]: check if we can rewrite this method as below , also after modifying this check for case when precision and scale having value 38.
private static int[] adjustPrecisionAndScaleIfNeeded(String dataType, int precision, int scale) {
    final int MAX_PRECISION_SCALE = 38;
    if ("BIGNUMERIC".equals(dataType)) {
        precision = Math.min(precision, MAX_PRECISION_SCALE);
        scale = Math.min(scale, MAX_PRECISION_SCALE);
    }
    return new int[] {precision, scale};
}

**Review by sauajmera — COMMENTED** [review]
given one review comment.

**Review by sauajmera — APPROVED** [review]
changes looks good to me.

File: core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java (hunk 1 above)
Comment (mithilagarud-dm) [reply core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java]: Done

**Review by dm-shantanuc — APPROVED** [review]

**Review by dm-shantanuc — APPROVED** [review]

**Pull request merged**